src/client/datascience/baseJupyterSession.ts

```diff
@@ -368,7 +370,11 @@ export abstract class BaseJupyterSession implements IJupyterSession {
         timeoutMS: number
     ): Promise<ISessionWithSocket>;
 
-    protected async waitForIdleOnSession(session: ISessionWithSocket | undefined, timeout: number): Promise<void> {
+    protected async waitForIdleOnSession(
+        session: ISessionWithSocket | undefined,
+        timeout: number,
+        isRestartSession?: boolean
```

Comment:
Whether this session is a restart session or not.
For instance if we're dealing with a Jupyter server or remote jupyter server, we can & should always kill restart sessions.